lightning/src/blinded_path/payment.rs

```diff
@@ -437,7 +424,7 @@ impl Readable for BlindedPaymentTlvs {
 			(12, payment_constraints, required),
 			(14, features, (option, encoding: (BlindedHopFeatures, WithoutLength))),
 			(65536, payment_secret, option),
-			(65537, payment_context, (default_value, PaymentContext::unknown())),
+			(65537, payment_context, option),
```

Comment:
Probably a silly question about the code here, but what happens if the `payment_context` is already set to `UnknownPaymentContext` in a running LDK node? 

Is it possible for this event to occur?

Reply:
Let me give this a shot!

Since we are removing the variant in this PR, an upgraded LDK node won't "understand" the `payment_context` it encounters during a read. As a result, parsing the `BlindedPaymentTlvs` and, by extension, that `Blinded Payment Path` will fail. However, this should be fine because the introduction of `authentication` in this PR inherently breaks legacy compatibility, meaning the path wouldn't be usable regardless.

@jkczyz, hope I got this right! :)

Reply:
Yeah, when receiving a payment we'd fail to parse any TLVs using `PaymentContext::unknown()`. I believe this should be rare as it would only occur for `BlindedPaymentPath`s that preexisted `PaymentContext`. Additionally, the `PaymentContext::Unknown` variant cannot be created outside of the crate, so there's no way for a user to construct one. Any `BlindedPaymentPath` containing one could be parsed, however, since that data is encrypted in the path. But we never created those. It's only used when parsing the TLVs from incoming payments.